pybliometrics/scopus/serial_search.py

```diff
@@ -1,49 +1,44 @@
-from collections import namedtuple, OrderedDict
+from collections import OrderedDict
 
 from pybliometrics.scopus.superclasses import Search
 
 
 class SerialSearch(Search):
     @property
     def results(self):
-        """A list of namedtuples representing results of serial search. The
-        number of fields may vary from one search query to another depending
-        on the length of yearly data Note: can be empty.
+        """A list of OrderedDicts representing results of serial search. The
+        number of keys may vary from one search result to another depending
+        on the length of yearly data. Note: can be empty.
         """
         out = []
-        search_results = self._json['serial-metadata-response'].get('entry')
-        if search_results:
-            # Namedtuple fields from union of keys of each dictionary in result list
-            fields = [_clean_field_name(i) for i in _form_fields(search_results)]
-            serial = namedtuple('Serial', fields)
-            for i in search_results:
-                # OrderedDict to populate with individual serial data
-                obs = OrderedDict.fromkeys(fields)
-                for key in i:
-                    if key != '@_fa' and i[key]:
-                        if key == 'subject-area':
-                            subject_data = _merge_subject_data(i[key])
-                            obs['subject_area_codes'] = subject_data[0]
-                            obs['subject_area_abbrevs'] = subject_data[1]
-                            obs['subject_area_names'] = subject_data[2]
-                        elif key == 'SNIPList' or key == 'SJRList':
-                            for j in _retrieve_source_rankings(i[key]):
+        search_results = self._json['serial-metadata-response'].get('entry', [])
+        for i in search_results:
+            # OrderedDict to populate with individual serial data
+            obs = OrderedDict()
+            for key in i:
+                if key != '@_fa' and i[key]:
```

Comment:
One can remove one level by inverting this statement followed by `continue`, i.e.
```
if key == '@_fa':
    continue
<normal code>
```
What's the purpose of testing `i[key]`? It's the element we're iterating over, isn't?

Reply:
Some keys can have a `None` value, hence testing.

Reply:
Here is an example of a result dictionary with keys with `None` values:
```python
{'@_fa': 'true',
 'dc:title': '10th International Symposium on Integrated Circuits, Devices and Systems, ISIC-2004: Integrated Systems on Silicon - Proceedings',
 'dc:publisher': None,
 'coverageStartYear': '2004',
 'coverageEndYear': '2004',
 'prism:aggregationType': 'conferenceproceeding',
 'source-id': '78598',
 'openaccess': None,
 'openaccessArticle': None,
 'openArchiveArticle': None,
 'openaccessType': None,
 'openaccessStartDate': None,
 'oaAllowsAuthorPaid': None,
 'subject-area': [{'@_fa': 'true',
   '@code': '2200',
   '@abbrev': 'ENGI',
   '$': 'Engineering (all)'}],
 'SNIPList': {'SNIP': [{'@_fa': 'true', '@year': '2007', '$': '0.032'}]},
 'SJRList': {'SJR': [{'@_fa': 'true', '@year': '2007', '$': '0.102'}]},
 'link': [{'@_fa': 'true',
   '@ref': 'scopus-source',
   '@href': 'https://www.scopus.com/source/sourceInfo.url?sourceId=78598'}],
 'yearly-data': None}
```
The most problematic part are keys with dictionary values (e.g. `yearly-data`). Normally, the `yearly-data` key from the above example would look like this:
```python
{...,
 'yearly-data': {'info': [{'@_fa': 'true',
    '@year': '1996',
    'publicationCount': '0',
    'citeCountSCE': '0',
    'zeroCitesSCE': '0',
    'zeroCitesPercentSCE': None,
    'revPercent': '0'},
   {'@_fa': 'true',
    '@year': '1997',
    'publicationCount': '0',
    'citeCountSCE': '0',
    'zeroCitesSCE': '0',
    'zeroCitesPercentSCE': None,
    'revPercent': '0'}]}}
```
Testing for existence of keys is not enough. Since keys with container values are passed through auxiliary functions, in this case we would be searching for `info` key using `result['yearly-data'].get('info')` which will return `AttributeError` on `None`. The solution is to also test value of each key.
Alternatively, we could test value of each key with container value but that would mean adding additional `if` clauses.

Reply:
I see. Then let's do like this:
``````